apps/dashboard/app/controllers/batch_connect/settings_controller.rb

```diff
@@ -0,0 +1,49 @@
+# The controller to manage BatchConnect saved settings
+class BatchConnect::SettingsController < ApplicationController
+  include BatchConnectConcern
+  include UserSettingStore
+
+  # GET /batch_connect/<app_token>/settings/<settings_name>
+  def show
+    set_app_groups
+    set_saved_settings
+
+    settings_params = settings_request_params
+    app_token = settings_params[:token]
+    settings_name = settings_params[:id]
+    settings_values = bc_templates(app_token).fetch(settings_name.to_sym, {})
+    @settings = BatchConnect::Settings.new(app_token, settings_name, settings_values)
+    if @settings.outdated?
+      flash.now[:alert] = t('dashboard.bc_saved_settings.outdated_message', app_title: @settings.app.title)
+    end
```

Comment:
I think there's a bug here where a user can edit the URL to a setting that doesn't exist. You get a page with the settings as if they're real, but they're not. Then if you press `Launch` obviously it doesn't work because there are no settings.

Curiously enough though, you can delete settings that don't exist - the systems seems to think that that's OK.

Reply:
will take a look

Reply:
Added a check for missing saved settings